version: '3.8'

services:
  db:
    image: postgres:12.4
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    environment:
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_DB: ${DB_NAME}
    restart: always
    healthcheck:
      test: pg_isready -U ${DB_USER} -d ${DB_NAME}
      interval: 5s
      timeout: 3s
      retries: 10

  run_migrations:
    image: ${DOCKERHUB_USER}/todolist:latest
    env_file: .env
    depends_on:
      db:
        condition: service_healthy
    command: python manage.py migrate

  api:
    image: ${DOCKERHUB_USER}/todolist:latest
    env_file: .env
    depends_on:
      db:
        condition: service_healthy
      run_migrations:
        condition: service_completed_successfully

  collect_static:
    image: ${DOCKERHUB_USER}/todolist:latest
    env_file: .env
    command: python manage.py collectstatic -c --no-input
    volumes:
      - django_static:/app/static/

  frontend:
    image: sermalenk/skypro-front:lesson-34
    restart: on-failure
    ports:
      - "80:80"
    depends_on:
      api:
        condition: service_started
      collect_static:
        condition: service_completed_successfully
    volumes:
      - ./deploy/nginx.conf:/etc/nginx/conf.d/default.conf
      - django_static:/usr/share/nginx/html/static/


volumes:
  postgres_data:
  django_static:
